instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/FieldMapping.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v2_2;
+
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+
+class FieldMapping {
+
+  enum Type {
+    REQUEST,
+    RESPONSE;
+  }
+
+  static FieldMapping request(String attribute, String fieldPath) {
+    return new FieldMapping(Type.REQUEST, attribute, fieldPath);
+  }
+
+  static FieldMapping response(String attribute, String fieldPath) {
+    return new FieldMapping(Type.RESPONSE, attribute, fieldPath);
+  }
+
+  FieldMapping(Type type, String attribute, String fieldPath) {
+    this.type = type;
+    this.attribute = attribute;
+    this.fieldPath = fieldPath;
+    this.fields = fieldPath.split("\\.");
```

Comment:
I wanted to avoid interpretation issues - with path it's obvious that "nested" fields are required, with array it would look like a bunch of fields on the same level. In other words - decided on readability vs performance gain here.